libsolutil/Assertions.h

```diff
@@ -35,9 +35,9 @@ namespace solidity::util
 #if defined(_MSC_VER)
 #define ETH_FUNC __FUNCSIG__
 #elif defined(__GNUC__)
-#define ETH_FUNC __PRETTY_FUNCTION__
+#define ETH_FUNC __FUNCTION__
 #else
-#define ETH_FUNC __func__
+#define ETH_FUNC __FUNCTION__
```

Comment:
If this really works I would argue that we don't need the function signature, because function name & line number should be sufficient...

Reply:
ok looks good.. @ekpyron `__PRETTY_FUNCTION__` seem to behave differently for `gcc` & `clang`, where `gcc` seem to add the full namespace info to function parameters, but `clang` not.. I think the easiest solution is to just not use `__PRETTY_FUNCTION__` here.. maybe we could just have a single define `#define ETH_FUNC __FUNCTION__` for all compilers?

Reply:
Aaah, so there's our culprit. We really shouldn't be relying on non-standard macros, at least for anything that is going to be checked and compared by the CI using different compilers. We can either roll our own, or just

Reply:
The mistake here really is making any of this part of "regular" compiler output - assertion violations, a.k.a. ICEs, are nothing that we usually expect to happen, but just safeguards. And for that we can output (during what's pretty much the compiler crashing) as much information as possible, including nicely printed function names - and since this is not proper compiler output but basically a crash report, it doesn't matter too much, if there is slight platform differences.

Reply:
A bit annoying now, though - we should just not have tested this :-D.
The main reason for calling this experimental is that it *doesn't* have proper error reporting - which we'll just get for free once we switch to json schemata.

But I'm not convinced by the assertion messages using ``__FUNCTION__`` really - but now that we've opened Pandora's Box here, we need to see how we can deal with this nicely - the platform dependency is of course generally, even for ICEs, not overly nice.

Reply:
Alright, compared to platform specific messages, the uglier function names are maybe the lesser evil.
But I'd say then we give it an additional changelog entry like a Bugfix/General "Change formatting of function names in assertion failure messages to avoid plattform differences."

Is the MSVC macro even guaranteed to behave the same? ``__FUNCSIG`` rather sounds like the full signature and not only the function name...

Reply:
I'm actually even tending towards just removing the tests and addressing the platform discrepancy here separately - it applies to all ICEs equally anyways.

Reply:
great! I just pushed everything